diff -u vdr-1.7.28/config.c vdr-1.7/config.c
--- vdr-1.7.28/config.c	2012-05-11 12:06:57.000000000 +0100
+++ vdr-1.7/config.c	2012-06-08 11:03:58.670418555 +0100
@@ -415,6 +415,7 @@
   UseSubtitle = 1;
   UseVps = 0;
   VpsMargin = 120;
+  VpsFallback = 0;
   RecordingDirs = 1;
   FoldersInTimerMenu = 1;
   NumberKeysForChars = 1;
@@ -610,6 +611,7 @@
   else if (!strcasecmp(Name, "UseSubtitle"))         UseSubtitle        = atoi(Value);
   else if (!strcasecmp(Name, "UseVps"))              UseVps             = atoi(Value);
   else if (!strcasecmp(Name, "VpsMargin"))           VpsMargin          = atoi(Value);
+  else if (!strcasecmp(Name, "VpsFallback"))         VpsFallback      = atoi(Value);
   else if (!strcasecmp(Name, "RecordingDirs"))       RecordingDirs      = atoi(Value);
   else if (!strcasecmp(Name, "FoldersInTimerMenu"))  FoldersInTimerMenu = atoi(Value);
   else if (!strcasecmp(Name, "NumberKeysForChars"))  NumberKeysForChars = atoi(Value);
@@ -708,6 +710,7 @@
   Store("UseSubtitle",        UseSubtitle);
   Store("UseVps",             UseVps);
   Store("VpsMargin",          VpsMargin);
+  Store("VpsFallback",        VpsFallback);
   Store("RecordingDirs",      RecordingDirs);
   Store("FoldersInTimerMenu", FoldersInTimerMenu);
   Store("NumberKeysForChars", NumberKeysForChars);
diff -u vdr-1.7.28/config.h vdr-1.7/config.h
--- vdr-1.7.28/config.h	2012-04-15 11:45:32.000000000 +0100
+++ vdr-1.7/config.h	2012-06-08 11:03:58.672418431 +0100
@@ -285,6 +285,7 @@
   int UseSubtitle;
   int UseVps;
   int VpsMargin;
+  int VpsFallback;
   int RecordingDirs;
   int FoldersInTimerMenu;
   int NumberKeysForChars;
diff -u vdr-1.7.28/menu.c vdr-1.7/menu.c
--- vdr-1.7.28/menu.c	2012-05-12 14:08:23.000000000 +0100
+++ vdr-1.7/menu.c	2012-06-08 11:03:58.675418246 +0100
@@ -3076,6 +3076,7 @@
   Add(new cMenuEditBoolItem(tr("Setup.Recording$Use episode name"),          &data.UseSubtitle));
   Add(new cMenuEditBoolItem(tr("Setup.Recording$Use VPS"),                   &data.UseVps));
   Add(new cMenuEditIntItem( tr("Setup.Recording$VPS margin (s)"),            &data.VpsMargin, 0));
+  Add(new cMenuEditBoolItem(tr("Setup.Recording$Use running status as VPS fallback"), &data.VpsFallback));
   Add(new cMenuEditBoolItem(tr("Setup.Recording$Mark instant recording"),    &data.MarkInstantRecord));
   Add(new cMenuEditStrItem( tr("Setup.Recording$Name instant recording"),     data.NameInstantRecord, sizeof(data.NameInstantRecord)));
   Add(new cMenuEditIntItem( tr("Setup.Recording$Instant rec. time (min)"),   &data.InstantRecordTime, 1, MAXINSTANTRECTIME));
diff -u vdr-1.7.28/timers.c vdr-1.7/timers.c
--- vdr-1.7.28/timers.c	2012-04-25 10:02:03.000000000 +0100
+++ vdr-1.7/timers.c	2012-06-08 11:03:58.676418184 +0100
@@ -431,7 +431,7 @@
   deferred = 0;
 
   if (HasFlags(tfActive)) {
-     if (HasFlags(tfVps) && event && event->Vps()) {
+     if (HasFlags(tfVps) && event && (Setup.VpsFallback || event->Vps())) {
         if (Margin || !Directly) {
            startTime = event->StartTime();
            stopTime = event->EndTime();
